library/macos/FLETextInputModel.h

```diff
@@ -51,4 +51,28 @@ typedef NS_ENUM(NSUInteger, FLETextAffinity) { FLETextAffinityUpstream, FLETextA
  */
 @property(nonnull) NSDictionary *state;
 
+/**
+ * Id of the text input client.
+ */
+@property(nonatomic, readonly) NSNumber *clientId;
+
+/**
+ * Keyboard type of the client. See available options:
+ * https://github.com/flutter/flutter/blob/0468742910c6e9230533e8e2c616520116b906e6/packages/flutter/lib/src/services/text_input.dart#L25
```

Comment:
Same comments about links.

(We should really look at whether we could reasonably share a single C++ implementation of the text model, rather than having two implementations. If you're interested in taking a look at that next while this code is fresh in your mind, it would be great.)

Reply:
Sounds good! I'll take a stab at that next.